### PR TITLE
add rabbitmq and more

### PR DESCRIPTION
**Pull request by sjoerdie**
- More flexible healthprobes for app and nginx, needed because openforms 403 difficulty;
- Add RabbitMQ deploy, service and vars (required for opennotifications);
- django_app_k8s_ingress_classname [Deprecating the Ingress Class Annotation ](https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/#deprecating-the-ingress-class-annotation);
- More flexible tls-secret, define tls-secret name and skip tls-secret deployment (when secrets are deployed by other methods (eg azure vault or cert-manager letsencrypt);

Tested and works.

### CODE REVIEW

File: roles/django_app_k8s/templates/rabbitmq-svc.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+
+kind: Service
+apiVersion: v1
+metadata:
+  labels:
+    app.kubernetes.io/name: rabbitmq
+    app.kubernetes.io/instance: "{{ django_app_k8s_instance }}"
+    app.kubernetes.io/version: "3.7"
```
Comment (sergei-maertens) [inline roles/django_app_k8s/templates/rabbitmq-svc.yml.j2:9]: I'd correlate that to the rabbitmq image tag - I see that latest is on 3.10 by now

File: roles/django_app_k8s/templates/rabbitmq.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,69 @@
+---
+
+kind: Deployment
+apiVersion: v1
+metadata:
+  labels:
+    app.kubernetes.io/name: rabbitmq
+    app.kubernetes.io/instance: "{{ django_app_k8s_instance }}"
+    app.kubernetes.io/version: "3.7"
```
Comment (sergei-maertens) [inline roles/django_app_k8s/templates/rabbitmq.yml.j2:9]: See service version remark

File: roles/django_app_k8s/templates/rabbitmq.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      app.kubernetes.io/name: rabbitmq
+      app.kubernetes.io/instance: "{{ django_app_k8s_instance }}"
+      app.kubernetes.io/part-of: "{{ django_app_k8s_group }}"
+  template:
+    metadata:
+      name: opennotificaties-worker
+      labels:
+        app.kubernetes.io/name: rabbitmq
+        app.kubernetes.io/instance: "{{ django_app_k8s_instance }}"
+        app.kubernetes.io/version: "3.7"
```
Comment (sergei-maertens) [inline roles/django_app_k8s/templates/rabbitmq.yml.j2:27]: I am repeating myself :P

File: roles/django_app_k8s/templates/rabbitmq.yml.j2 [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        app.kubernetes.io/component: async-workers
+        app.kubernetes.io/part-of: "{{ django_app_k8s_group }}"
+        app.kubernetes.io/managed-by: Ansible
+{% for label in django_app_k8s_extra_rabbitmq_labels %}
+        {{ label }}: {{ django_app_k8s_extra_rabbitmq_labels[label] }}
+{% endfor %}                
+    spec:
+      containers:
+      - name: rabbitmq
+        image: rabbitmq:3.7
```
Comment (sergei-maertens) [inline roles/django_app_k8s/templates/rabbitmq.yml.j2:37]: maybe a var to control the rabbitmq image/version?

File: roles/django_app_k8s/templates/rabbitmq.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          tcpSocket:
+            port: 5672
+          initialDelaySeconds: 5
+          periodSeconds: 10
+
+        livenessProbe:
+          tcpSocket:
+            port: 5672
+          initialDelaySeconds: 15
+          periodSeconds: 20
```
Comment (sergei-maertens) [inline roles/django_app_k8s/templates/rabbitmq.yml.j2:54]: I've seen RabbitMQ go down because of memory watermarks - if 90% of available memory is used, the channels are closed and it basically becomes inoperable. Maybe we can see if we can make this probe better so that we get hard crashes and monitoring starts screaming if it becomes unavailable? That could be for a separate PR though.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: roles/django_app_k8s/templates/rabbitmq.yml.j2 (hunk 5 above)
Comment (sjoerdie) [reply roles/django_app_k8s/templates/rabbitmq.yml.j2:54]: Ok, separate PR it is.

File: roles/django_app_k8s/templates/rabbitmq.yml.j2 (hunk 4 above)
Comment (sjoerdie) [reply roles/django_app_k8s/templates/rabbitmq.yml.j2]: added and using: `django_app_k8s_rabbitmq_version: "3.10"`

File: roles/django_app_k8s/templates/rabbitmq.yml.j2 (hunk 2 above)
Comment (sjoerdie) [reply roles/django_app_k8s/templates/rabbitmq.yml.j2]: added and using: `django_app_k8s_rabbitmq_version: "3.10"`

File: roles/django_app_k8s/templates/rabbitmq-svc.yml.j2 (hunk 1 above)
Comment (sjoerdie) [reply roles/django_app_k8s/templates/rabbitmq-svc.yml.j2]: added and using: `django_app_k8s_rabbitmq_version: "3.10"`

**Review by sergei-maertens — APPROVED** [review]
